core/server/proxy/src/main/java/alluxio/proxy/S3RestServiceHandler.java

```diff
@@ -0,0 +1,116 @@
+@Produces(MediaType.APPLICATION_XML)
+@Consumes(MediaType.APPLICATION_XML)
+public final class S3RestServiceHandler {
+  public static final String SERVICE_PREFIX = "s3";
+
+  /**
+   * Bucket must be a directory directly under the root directory or a mount point.
+   * If bucket is a nested directory under a mount point, it must be in the format like
+   * mount:point:bucketname where : is used as path separator instead of /.
+   */
+  public static final String BUCKET_SEPARATOR = ":";
+  public static final String PATH_SEPARATOR = "/";
+
+  // Bucket is the first component in the URL path.
+  public static final String BUCKET_PARAM = "{bucket}/";
+
+  private final FileSystem mFileSystem;
+
+  /**
+   * Constructs a new {@link S3RestServiceHandler}.
+   *
+   * @param context context for the servlet
+   */
+  public S3RestServiceHandler(@Context ServletContext context) {
+    mFileSystem =
+        (FileSystem) context.getAttribute(ProxyWebServer.FILE_SYSTEM_SERVLET_RESOURCE_KEY);
+  }
+
+  /**
+   * @summary creates a bucket
+   * @param bucket the Alluxio path
+   * @return the response object
+   */
+  @PUT
+  @Path(BUCKET_PARAM)
+  @ReturnType("java.lang.Void")
+  public Response createBucket(@PathParam("bucket") final String bucket) {
+    return S3RestUtils.call(bucket, new S3RestUtils.RestCallable<Void>() {
+      @Override
+      public Void call() throws S3RestUtils.S3Exception {
+        String bucketName = PATH_SEPARATOR + bucket;
+        if (bucketName.contains(BUCKET_SEPARATOR)) {
+          bucketName = bucketName.replace(BUCKET_SEPARATOR, PATH_SEPARATOR);
+          // Assure that the bucket is directly under a mount point.
+          AlluxioURI parent = new AlluxioURI(bucketName).getParent();
+          try {
+            if (!mFileSystem.getMountTable().containsKey(parent.getPath())) {
+              throw new S3RestUtils.S3Exception(bucketName,
+                  S3RestUtils.ErrorCode.INVALID_NESTED_BUCKET_NAME);
+            }
+          } catch (IOException | AlluxioException e) {
+            throw new S3RestUtils.S3Exception(e, bucketName, S3RestUtils.ErrorCode.INTERNAL_ERROR);
+          }
+        }
+
+        // Create the bucket.
+        CreateDirectoryOptions options = CreateDirectoryOptions.defaults();
+        options.setWriteType(Configuration.getEnum(PropertyKey.PROXY_S3_WRITE_TYPE,
+              WriteType.class));
+        try {
+          mFileSystem.createDirectory(new AlluxioURI(bucketName), options);
+        } catch (InvalidPathException e) {
+          throw new S3RestUtils.S3Exception(bucketName, S3RestUtils.ErrorCode.INVALID_BUCKET_NAME);
+        } catch (FileAlreadyExistsException e) {
+          throw new S3RestUtils.S3Exception(bucketName,
+              S3RestUtils.ErrorCode.BUCKET_ALREADY_EXISTS);
+        } catch (IOException | AlluxioException e) {
+          throw new S3RestUtils.S3Exception(e, bucketName, S3RestUtils.ErrorCode.INTERNAL_ERROR);
+        }
```

Comment:
Nevermind, it might be better to do it explicitly if we cannot do something like
```
try {
  ...
} catch ...
  throw toS3Exception(e)
}